src/security/PeerOptions.cc

```diff
@@ -284,6 +291,9 @@ Security::PeerOptions::createClientContext(bool setOptions)
     if (t) {
         if (setOptions)
             updateContextOptions(t);
+
+        Must(updateContextCiphers(t)); // ciphers not working is fatal for connections to server/peer
```

Comment:
Since most of that behaviour change is in the ssl-bump logic I cannot test, I am not touching it myself at this point. If you or Christos want to do the improvement as a followup PR with appropriate testing that is welcome.

As far as I understand the ServerOptions context creation affects both config parsing and cert mimic context creation. Each of those has different handling, and ciphers which may be valid for the configured cert/key etc may not be valid for run-time generated contexts. So in that case the bool result is used to inform the caller it has to do its variable failover action.

The PeerOptions::updateContextCiphers() method is the single place ciphers are "parsed" / added to the context - shared by both client and server context types. So right now it cannot throw until all of the caller logics are converted to handling exceptions instead of expecting the true/false result.

(The above Must() is possible because the PeerOptions:createClientContext is only used during config parse / init time.)

Reply:
> are we okay on the updateContextCiphers() situation

I copied your question to this change request because I assume that you are talking about this change request when asking about "the updateContextCiphers() situation". If my assumption is wrong, please let me know. 

I consider this change request addressed.

I continue to urge you to stop creating significant changes to the code that you refuse to test. I understand that some of those changes are side effects of the projects you are working on, but that fact is immaterial. No changes at all would be much better than half-baked, half-tested changes combined with refused change requests.

> so this can be merged?

I do not know. Did you address all the other change requests? I am waiting for your signal that the ball is on my side again. Please note that I am OoO until Wednesday though.

> or you strongly wanting all the callers to have try-catch added for Must() ?

I have never wanted that.

Reply:
Okay, I think it is ready for commit then modulo anything after the changes. marking for review.